app/services/build_heritage.rb

```diff
@@ -16,7 +16,16 @@ def convert_params_for_model(new_params)
         service[:port_mappings_attributes] = service.delete(:port_mappings) if service[:port_mappings].present?
 
         unless service[:listeners].nil?
-          listener_map = Hash[Endpoint.where(name: service[:listeners].map { |e| e[:endpoint] }).pluck(:name, :id)]
+          endpoint_names = service[:listeners].map { |e| e[:endpoint] }
+
+          endpoints = if @district.present?
+            Endpoint.where(name: endpoint_names, district: @district).pluck(:name, :id)
+          else
+            Endpoint.where(name: endpoint_names).pluck(:name, :id)
+          end
```

Comment:
Hmm this would still cause the same issue. Since barcelona is open source if someone that used it made an endpoint called `staging-blue-green` in the past, and now made a new district called `staging-blue` but forgot to make an endpoint called `green` barcelona will choose the old one!

Reply:
@davidsiaw 
I think this test shows it works in such a case
https://github.com/degica/barcelona/pull/660/commits/6322a4682ce1a05285048788162111ace32d30c1